### DISCUSSION TITLE
Can't get the developer page to respond on anything but the Unit ID

### DISCUSSION THREAD

**Discussion by 11sas11** [amberelectric/public-api#83] [Q&A]
I am a completely new at this stuff so please excuse me of send me off to various references to sort it out.  But I can't even get the web page to work properly.  I can login and get my ID okay, but if I try and get prices, it just doesn't work.  Any ideas.
![image](https://user-images.githubusercontent.com/48985019/138634025-27f098e6-5159-4dbb-973c-37b3728735a0.png)

**Comment by J-Rod-16** [comment]
I'm having the same issue

**Reply by J-Rod-16** [reply]
I solved my issue, incorrect date period. looks to be different from yours

**Comment by madpilot** [comment]
Sorry for the delay in responding - I must have missed the notification.

Are you in OSX or Linux? If you cut and paste that cURL command in to a terminal, then add -v it will give you a better error

**Comment by madpilot** [comment]
Wait. I see. You need to copy the site ID from the first request (01FJ8…) into the siteId field in the prices query. 

The library we use to display the API isn't smart enough to prefill it

**Reply by 11sas11** [reply]
I did copy and paste it.  Does it need quotation marks around it?  I tried it both ways and it didn't work.  I am using Windows 10 and Chrome browser.  (I want to use it in C# .NET - Visual Studio), but was having problems so tried it on the site to get it to work.  Current price is all I need - same issue.  The only one that I get a response for is SiteID.

**Reply by madpilot** [reply]
I just tried it, and it's working here

![image](https://user-images.githubusercontent.com/19585/140258266-e7dad1c3-7961-47ca-ae6c-951211dd5f5f.png)

![image](https://user-images.githubusercontent.com/19585/140258285-c0eeb747-c6ba-4859-9a47-625dea5602e4.png)

That screenshot you have posted definitely has the wrong site id.

**Reply by madpilot** [reply]
If you open dev tools in chrome you can possibly get a better error message (The CORS one is a generic one the browser returns)

**Reply by 11sas11** [reply]
Yes, that screen shot had the incorrect ID, but I had tried it multiple times with the correct ID and quotes etc.  But anyway checked now and got, tbh I just wanted to check I was entering the right data in the right field which is why I would try it out before coding.  Your example shows me anyway.  So no need to get this working now.
![image](https://user-images.githubusercontent.com/48985019/140450670-08338f15-fb6f-451b-a9ec-0999fe76e84f.png)

**Reply by madpilot** [reply]
Ahhh - you had the auth key in the `site_id` field.

BTW: For security, I'd delete that key and create a new one.

**Reply by 11sas11** [reply]
yes, good point.  thx

**Reply by 11sas11** [reply]
um okay, I am sure I had the ID, but if I am that much of an idiot, thanks for bearing with me.  I tried again and the current price does not return anything but the 30 sample one is returning correctly.

**Reply by madpilot** [reply]
Would it help to do a 15 minute video call next week?

I can walk you through it

**Reply by 11sas11** [reply]
Thanks for the offer.  I will have a crack this weekend, and if I am still struggling I might take you up on it.  Btw the price is only import right?  I didn't see a filed to select the identifier or type?

**Reply by madpilot** [reply]
If you have solar you'll see a feedIn channel which has the export price

**Reply by 11sas11** [reply]
@madpilot I got it done.  Actually my first open API application - so a bit of a steep learning curve for this amateur.  I can read the data.  I do note that if I send forward 0, and rearward 0, I get a null response, but if I set either to 1 I get the current and either a past or future set.  Now I have the data - time to start using it ;-)  - I really don't know what I was doing wrong before, thank you for bearing with me.
![image](https://user-images.githubusercontent.com/48985019/140607397-c375f646-5b88-4b95-97f8-b8306b5dea40.png)

**Reply by madpilot** [reply]
Woohoo!

I must admit setting prev and next to 0 was not a use-case I had anticipated... I will do something about that on Monday.

Congratulations, though!
